docs/flow/modin/data_management/functions.rst

```diff
@@ -0,0 +1,127 @@
+Function Module Description
+"""""""""""""""""""""""""""
+
+Brief description
+'''''''''''''''''
+Most of the functions that are evaluated by `QueryCompiler` can be categorized into
+one of the patterns: Map, MapReduce, Binary functions, Fold functions, etc. Function
+module provides templates to easily build such types of functions. These templates
+are supposed to be used at the `QueryCompiler` level since each built function accepts
+and returns `QueryCompiler`.
+
+High-Level Module Overview
+''''''''''''''''''''''''''
+Each template represented with a class with the corresponding name and implements
+``register`` method, which takes functions to apply in an appropriate way and
+instantiate the related template. Functions that are passed to the ``register``
+will be executed under deserialized and preprocessed (depends on the template)
+partitions, so the function would take one of the pandas object: ``pandas.DataFrame``,
+``pandas.Series`` or ``pandas.DataFrameGroupbyObject``.
+
+.. note:: 
+    Currently, functions that are built in that way are supported only in a pandas
+    backend (can be used only in `PandasQueryCompiler`).
+
+Function module provides templates for this type of function:
+
+Map functions
+-------------
+Apply function to each partition in parallel. **Note**: map function should not change the shape of the partitions.
+
+.. figure:: /img/map_evaluation.svg
+    :align: center
+
+Reduction functions
+-------------------
+Function that reduces specified axis into a scalar, but requires knowledge about the whole axis.
+Be aware that providing this knowledge may be expensive because the execution engine has to
+concatenate partitions along the specified axis. Also, note that the execution engine expects
```

Comment:
The note here is specific to our current implementation, not necessarily general. You can add it as a `.. note:` and specify that this is the case for the `PandasQueryCompiler`.

Reply:
I see. It is specific to pandas backend, but that is fine.